src/core/agent/services/identifierService.ts

```diff
@@ -152,10 +153,16 @@ class IdentifierService extends AgentService {
     const metadata = await this.identifierStorage.getIdentifierMetadata(
       identifier
     );
+    const connectedDApp =
+      PeerConnection.peerConnection.getConnectedDAppAddress();
+    const peerConnectingAid = PeerConnection.peerConnection.getConnectingAid();
     this.validArchivedIdentifier(metadata);
     await this.identifierStorage.updateIdentifierMetadata(identifier, {
       isDeleted: true,
     });
+    if (connectedDApp !== "" && metadata.id === peerConnectingAid) {
```

Comment:
Is it `""` or just `undefined` by default?

I also think I might be cleaner with:
```
    if (
      PeerConnection.peerConnection.getConnectedDAppAddress() !== undefined &&
      metadata.id === PeerConnection.peerConnection.getConnectingAid()
    ) {
```

Just because I want to keep the peer connect things as non-intrusive as possible in these files so it's easier to refactor later in case we make peer connect an optional module, we will need a way to dynamically load this logic.

Reply:
It is `""` at the moment. The value comes from `connectedDAppAdress` in `src/core/cardano/walletConnect/peerConnection.ts`.

Reply:
OK cool.

We can still do:
```
if ( PeerConnection.peerConnection.getConnectedDAppAddress() !== "" && metadata.id === PeerConnection.peerConnection.getConnectingAid() ) {
```
for reasons mentioned above